sympy/assumptions/satask.py

```diff
@@ -332,7 +334,10 @@ def get_all_relevant_facts(proposition, assumptions, context,
 
     if use_known_facts:
         known_facts_CNF = CNF()
-        known_facts_CNF.add_clauses(get_all_known_facts())
+        if any(any(arg.kind == MatrixKind(NumberKind) for arg in pred.arguments) for pred in proposition.all_predicates()):
+            known_facts_CNF.add_clauses(get_all_known_matrix_facts())
+        else:
+            known_facts_CNF.add_clauses(get_all_known_number_facts())
```

Comment:
I don't know the new assumptions so well. What would this do if there were some predicates involving matrices and some others involving numbers?

Reply:
In that case it looks like it would break: it would use matrix facts only and not use any of the other predicates. 

It's the the following for loop where each expression should be checked to see if it's a matrix or not:
https://github.com/sympy/sympy/blob/a76b02fcd3a8b7f79b3a88df50c19eb7aee33c17/sympy/assumptions/satask.py#L350-L352

Of course, you'd need to make some other changes as well to the logic to make everything work.

Reply:
I ended up having to make a second loop since `kf_encoded` is built from `knownfacts` prior to the loop that you pointed out on line 350